src/main/java/com/linecorp/armeria/client/ClientBuilder.java

```diff
@@ -46,12 +48,18 @@
 
     private RemoteInvokerFactory remoteInvokerFactory = RemoteInvokerFactory.DEFAULT;
     private Function<Client, Client> decorator;
+    private String groupName;
 
     /**
      * Creates a new {@link ClientBuilder} that builds the client that connects to the specified {@code uri}.
      */
     public ClientBuilder(String uri) {
         this(URI.create(requireNonNull(uri, "uri")));
+
+        if (this.uri.getAuthority().equals(this.uri.getHost()) &&
+                ServerGroupRegistry.get(this.uri.getHost()) != null) {
+            this.groupName = this.uri.getHost();
```

Comment:
The port is unnecessary.we will replace the whole "serverGroup:myGroupName:port" to the real service address(already contains port info),right?